PurchasesTests/Purchasing/__Snapshots__/ProductInfoTests/testEncoding.1.json

```diff
@@ -1,26 +1,26 @@
 {
   "currency" : "UYU",
-  "introductory_price" : 0,
+  "introductory_price" : "15.13",
   "normal_duration" : "P3Y",
   "offers" : [
     {
       "offer_identifier" : "offerid1",
       "payment_mode" : 0,
-      "price" : 11
+      "price" : "11.2"
     },
     {
       "offer_identifier" : "offerid2",
       "payment_mode" : 1,
-      "price" : 12
+      "price" : "12.2"
     },
     {
       "offer_identifier" : "offerid3",
       "payment_mode" : 2,
-      "price" : 13
+      "price" : "13.3"
     }
   ],
   "payment_mode" : 1,
-  "price" : 49.99,
+  "price" : "49.99",
```

Comment:
To summarize the issue and suggested solution: changing our encoding from `NSDecimalNumber`'s implicit conversion to `AnyObject`, to `Decimal`'s `Encodable` implementation meant that decimal numbers got encoded as something like `11.199999999999999`.

After talking about it with @tonycosentini, he mentioned that the server stores the prices as strings, so that could be problematic. Considering that, and the fact that we don't want to send arbitrary amounts of precision and deal with floating point issues later on, I think it's best to encode the prices as `String`.
I've tested a purchase on my device and everything works, but let me know what you think @taquitos.

Reply:
as long as the backend can correctly parse the price as string I think that's fine, have you double-checked with the dashboard?

Reply:
I've tested a purchase, but I haven't checked the dashboard.

Reply:
<img width="717" alt="Screen Shot 2021-12-29 at 13 50 50" src="https://user-images.githubusercontent.com/685609/147706108-4fb250de-73d3-4311-9202-964cda8ef48a.png">